ui/prometheus-plugin/src/model/time.ts

```diff
@@ -33,11 +31,15 @@ export interface PrometheusTimeRange {
  * Get the time range for the current dashboard, converted to Prometheus time.
  */
 export function useDashboardPrometheusTimeRange() {
-  const timeRange = useDashboardTimeRange();
+  const { start, end } = useDashboardTimeRange();
 
-  // Only recalculate the time range if the value on the dashboard changes,
-  // otherwise relative time ranges will change every time this hook is called
-  return useMemoized(() => toPrometheusTimeRange(timeRange), [timeRange]);
+  // Only recalculate the time range if the value on the dashboard changes
+  return useMemoized(() => {
+    return {
+      start: Math.ceil(getUnixTime(start)),
+      end: Math.ceil(getUnixTime(end)),
```

Comment:
Don't we think this thing should be at the  `core` level / global level ? I don't see why it is related to Prometheus (particularly).

Reply:
I feel like unless we end up with multiple data source that deal with things in unix timestamp seconds (and not ms or some other date format), this is definitely Prometheus-specific. In Grafana, for example, this logic is Prom-specific and in the Prom DataSource, not in the core libraries.